pgxn/neon/walproposer_pg.c

```diff
@@ -399,6 +414,13 @@ nwp_shmem_startup_hook(void)
 	WalproposerShmemInit();
 }
 
+WalproposerShmemState *
+GetWalpropShmemState()
+{
+	Assert(walprop_shared != NULL);
+	return walprop_shared;
+}
+
 static WalproposerShmemState *
 walprop_pg_get_shmem_state(WalProposer *wp)
```

Comment:
This function is the same as `GetWalpropShmemState`, probably we don't need `GetWalpropShmemState` anymore?

Reply:
Not quite, `walprop_pg_get_shmem_state` takes `WalProposer` as input, while `GetWalpropShmemState` doesn't and assumes a global WalProposer. I could remove the `Assert`, but maybe it's better to keep both versions?

Reply:
(The issue is that walsender doesn't have a WalProposer, so it can't provide a non-null pointer).